main.js

```diff
@@ -834,6 +834,20 @@ function extendChannel(nodeID, comClass, valueId) {
 
     // Create channel
     if (objects[channelID]) {
+        // channel exists - make sure we don't have duplicate states
+        if (!nodes[nodeID].ready) {
+            for (var i in objects) {
+                if (!objects.hasOwnProperty(i)) continue;
+                if (i.startsWith(channelID + '.')) {
+                    if (objects[i].native && objects[i].native.instance === valueId.instance && objects[i].native.index === valueId.index &&
+                    objects[i].common.name !== valueId.label) {
+                        adapter.log.info('remove obsolete state:' + objects[i]._id);
+                        delObjects([objects[i]]);
```

Comment:
I would prefer if you add the objects that should be deleted to an array and call delObjects once after the for loop.

Reply:
Sure, no problem. Although I am actually thinking if I can break the loop when I found an entry as I can't imagine that more than one duplicate should exist.

Reply:
Hmm you're right. One valueID means one state. So the current solution plus `break` is ok with me.

Reply:
makes sense.
When this is done on initialization I mostly do it with getting all already existing states, remove all that I create from this list, so at the end the list contains all "not needed anymore" states and I simply kill them. I do not know if this would work here too ...

Reply:
@Apollon77 the tricky part is that I need node id, instance, index and the label to identify which state is correct. And I only have that information once all values have come in. I will probably move that cleanup to a separate function but I don't see a chance to have it called just once. I need to collect the correct values anyhow so I can also do the cleanup at the same time.

Reply:
I have modified the patch:
- moved to a separate function
- only called from value added as that reduces load
I would need someone with old values to test it though, for me it works....
@Apollon77 @AlCalzone

Reply:
Ok then yes makes sense how it is now separate function even better.
I can not test right now, but code seems ok